src/apify_client/clients/resource_clients/dataset.py

```diff
@@ -0,0 +1,377 @@
+import io
+from typing import Any, Dict, Generator, List, Optional, cast
+
+from ..._types import JSONSerializable
+from ..base.resource_client import ResourceClient
+
+
+class DatasetClient(ResourceClient):
+    """Sub-client for manipulating a single dataset."""
+
+    def __init__(self, *args: Any, **kwargs: Any) -> None:
+        """Initialize the DatasetClient."""
+        super().__init__(*args, resource_path='datasets', **kwargs)
+
+    def get(self) -> Optional[Dict]:
```

Comment:
(applies to all the other clients like this too)
Maybe we could override the [] operator and allow extracting these values directly using the DatasetClient instance? It would be a little more user friendly and wouldn't require much additional work. We could also easily do the same for the setter use-case of [].

Reply:
I think this would be wrong conceptually - when you call `client.dataset('someDatasetId')`, you're getting another client, not the dataset itself. Implementing `[]` would mix these two concepts together, and I think in the end it would be just more confusing to users.

Also, we'd have to download the whole resource again and again for every property we would be accessing. I think that would be weird as well, when you call a function like `.get()`, you kinda intuitively expect that it might take some time, but when you access something via the `[]` operator, you intuitively expect it to be already there and accessible quickly.

We can revisit this later, but for this first version I'd skip it and stick just to the functions.